classes/privacy/provider.php

```diff
@@ -0,0 +1,105 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Contains class mod_accredible\privacy\provider
+ *
+ * @package    mod_accredible
+ * @subpackage accredible
+ * @copyright  Accredible <dev@accredible.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_accredible\privacy;
+use core_privacy\local\metadata\collection;
+use core_privacy\local\request\approved_contextlist;
+use core_privacy\local\request\approved_userlist;
+use core_privacy\local\request\contextlist;
+use core_privacy\local\request\userlist;
+/**
+ * Ad hoc task that performs the actions for approved data privacy requests.
+ */
+class provider implements
+  \core_privacy\local\metadata\provider,
+  \core_privacy\local\request\plugin\provider,
+  \core_privacy\local\request\core_userlist_provider {
+
+    /**
+     * Returns meta data about this system.
+     *
+     * @param  collection $collection The collection to add metadata to.
+     * @return collection $collection The array of metadata.
+     */
+    public static function get_metadata(collection $collection) : collection {
+
+        $collection->add_external_location_link('accredible', [
+            'email' => 'privacy:metadata:accredible:email',
```

Comment:
We also use the grade right? That would be personal data under their definition, so would need to be included in the implementation.

Reply:
I added quiz grade to the metadata: https://github.com/accredible/moodle-mod_accredible/pull/58/commits/5708a043f6ee638114651f280446df10a3cffae8

![Screen Shot 2022-04-20 at 10 10 24](https://user-images.githubusercontent.com/20009662/164126989-c691e63d-386c-4b79-8ce6-938283b609df.png)